src/vaults/BalanceSheet.sol

```diff
@@ -126,27 +141,48 @@ contract BalanceSheet is Auth, Recoverable, IBalanceSheet, IBalanceSheetGatewayH
         uint128 amount
     ) external authOrManager(poolId) {
         AssetId assetId = poolManager.assetToId(asset, tokenId);
-        _withdraw(poolId, scId, assetId, asset, tokenId, receiver, amount);
+        IPoolEscrow escrow_ = escrow(poolId);
+        escrow_.withdraw(scId, asset, tokenId, amount);
+
+        D18 pricePoolPerAsset_ = _pricePoolPerAsset(poolId, scId, assetId);
+        emit Withdraw(poolId, scId, asset, tokenId, receiver, amount, pricePoolPerAsset_);
+
+        if (queueEnabled[poolId][scId]) {
+            queuedAssets[poolId][scId][assetId].decrease += amount;
+        } else {
+            sender.sendUpdateHoldingAmount(poolId, scId, assetId, amount, pricePoolPerAsset_, false);
+        }
+
+        escrow_.authTransferTo(asset, tokenId, receiver, amount);
     }
 
     /// @inheritdoc IBalanceSheet
-    function issue(PoolId poolId, ShareClassId scId, address to, uint128 shares) external authOrManager(poolId) {
-        _issue(poolId, scId, to, shares);
+    function issue(PoolId poolId, ShareClassId scId, address to, uint128 shares) public authOrManager(poolId) {
+        emit Issue(poolId, scId, to, _pricePoolPerShare(poolId, scId), shares);
+
+        if (queueEnabled[poolId][scId]) {
+            queuedShares[poolId][scId].increase += shares;
+        } else {
+            sender.sendUpdateShares(poolId, scId, shares, true);
+        }
+
+        IShareToken token = poolManager.shareToken(poolId, scId);
+        token.mint(to, shares);
     }
 
     /// @inheritdoc IBalanceSheet
     function revoke(PoolId poolId, ShareClassId scId, address from, uint128 shares) external authOrManager(poolId) {
-        _noteRevoke(poolId, scId, from, shares);
-        _executeRevoke(poolId, scId, from, shares);
-    }
+        emit Revoke(poolId, scId, from, _pricePoolPerShare(poolId, scId), shares);
 
-    /// @inheritdoc IBalanceSheet
-    /// @dev This function is mostly useful to keep higher level integrations CEI adherent.
-    function noteRevoke(PoolId poolId, ShareClassId scId, address from, uint128 shares)
```

Comment:
@mustermeiszer I don't think `noteRevoke` can work, since it would require the manager to be able to burn shares aftewrads, which is not possible. I also think for CEI compatibility, managers should just be able to call `revoke` at the end of their tx. I would remove it, do you agree?

Reply:
Good point! Then lets remove the `from` from the `revoke` too, like for deposits and assume the manager is the holder. Wdyt?

Reply:
Good point!

Reply:
This was actually a bit tricky, since it meant we need to first transfer shares to `asyncRequestManager` in `revokedShares` before they can be revoked. Which means `asyncRequestManager` needs to be endorsed.